examples/dummy_plugin/dummy_plugin/custom_validators.py

```diff
@@ -21,4 +24,21 @@ def clean(self):
             self.validation_error({"name": "Site name must be something valid"})
 
 
-custom_validators = [SiteCustomValidator]
+class RelationshipAssociationCustomValidator(PluginCustomValidator):
+    model = "extras.relationshipassociation"
+
+    def clean(self):
+        """
+        Custom validator for RelationshipAssociation to enforce that an IP Address(destination) must be
+        within the host range of a Prefix(source)
+        """
+        obj = self.context["object"]
+        if ContentType.objects.get_for_model(obj.source) == ContentType.objects.get_for_model(Prefix):
```

Comment:
We might need to think about the design implications here - for example, if there are multiple different Relationships defined that involve Prefixes, how should we scope the custom validator to only apply to RelationshipAssociations for a specific Relationship?

The best I can come up with at the moment is hard-coding a check for the relationship slug, i.e.:

```python
if obj.relationship.slug != "test-relationship":
    # Not a relationship we have an interest in validating
    return
```